test/basic/test_utils.jl

```diff
@@ -17,6 +17,16 @@ y_train_regression = rand(1000)
 end
 
 
+@testset "stringifyparams -- multiple calls won't mutate fields" begin
```

Comment:
These tests also pass on the existing LightGBM without your change because the `indices` are being passed to `categorical_feature` so each time you call `stringifyparams` the values in the `indices` (assigned to `categorical_features`) will be also mutated, hence the test will always be true even without your change.

The scenario to be considered based on the #133 should be following:
```
categorical_features = [1, 3, 5, 7, 9]
expected_indices = [0, 2, 4, 6, 8] #as there's `-1` subtraction to match the C's zero-based indices.
classifier = LightGBM.LGBMClassification(categorical_feature = categorical_features)
LightGBM.stringifyparams(classifier; verbosity=-1)
@test expected_indices == classifier.categorical_feature

LightGBM.stringifyparams(classifier; verbosity=-1)
@test expected_indices == classifier.categorical_feature
```
With your current change the indices are not corrected and they remain [1, 3, 5, 7, 9] not [0, 2, 4, 6, 8] so this also needs to be revised.

This test sis better placed in `test_fit.jl` rather than utils as the `stringifparams` function is part of the `fit.jl` and the existing tests are in `test_fit.jl`